app/src/main/java/ch/epfl/sweng/SDP/Matchmaker.java

```diff
@@ -0,0 +1,92 @@
+package ch.epfl.sweng.SDP;
+
+import android.util.Log;
+
+import com.google.firebase.database.*;
+import com.google.firebase.auth.*;
+
+import java.security.PublicKey;
+import java.util.HashMap;
+
+public class Matchmaker {
+
+
+    private static Matchmaker single_instance = null;
+    // static method to create instance of Singleton class
+    private DatabaseReference myRef;
+
+    /**
+     *
+     * @return returns a singelton instance.
+     */
+    public static Matchmaker getInstance()
+    {
+        if (single_instance == null) {
+            single_instance = new Matchmaker();
+        }
+
+        return single_instance;
+    }
+
+
+    /**
+     *  Matchmaker init.
+     */
+    public Matchmaker() {
+
+        FirebaseDatabase database = FirebaseDatabase.getInstance();
+        myRef = database.getReference("rooms");
+
+        // Read from the database
+        myRef.addValueEventListener(new ValueEventListener() {
```

Comment:
Here addListenerForSingleValueEvent() could be better: it automatically removes the listener after the call (and you use it just once)

Reply:
What do you mean? :)

Reply:
addValueEventListener() needs to be removed when you don't want to listen to changes anymore; addListenerForSingleValueEvent() does this automatically since it assumes you need the listener just once, which I think is the case here.